core/governance/engine.go

```diff
@@ -1036,6 +1039,26 @@ func (e *Engine) validateCancelGovernanceTransfer(transferID string) (types.Prop
 	return types.ProposalErrorUnspecified, nil
 }
 
+func (e *Engine) validateMarketUpdateState(update *types.MarketStateUpdateConfiguration) (types.ProposalError, error) {
+	marketID := update.MarketID
+	if !e.markets.MarketExists(marketID) {
+		e.log.Debug("market does not exist", logging.MarketID(marketID))
+		return types.ProposalErrorInvalidMarket, ErrMarketDoesNotExist
+	}
+
+	marketState, err := e.markets.GetMarketState(marketID)
+	if err != nil {
+		return types.ProposalErrorInvalidMarket, err
+	}
+
+	// if the market is already terminated or not yet started or settled
+	if marketState == types.MarketStateTradingTerminated || marketState == types.MarketStateSettled || marketState == types.MarketStateProposed {
+		return types.ProposalErrorInvalidMarket, ErrMarketStateUpdateNotAllowed
+	}
+
```

Comment:
I don't think we need to do anything - if a proposal for termination gets enacted, it will get the market into a terminal state and then any further proposal getting pre-enacted will fail here and rejected.

Reply:
What about the case where I submit a update-proposal before its the first one has been enacted? So that market hasn't changed state yet.

Reply:
still, the check on VerifyUpdateMarketState would fail the proposal before it gets enacted. 

```
state := market.GetMarketState()
if state == types.MarketStateCancelled || state == types.MarketStateClosed || state == types.MarketStateRejected || state == types.MarketStateSettled || state == types.MarketStateTradingTerminated {
			return fmt.Errorf("invalid state update request. Market is already in a terminal state")
		}
```

Reply:
Yea you're right, Cool.